File: app/src/main/java/com/example/learningverbs/utils/StringUtils.java
package com.example.learningverbs.utils;

import android.util.Patterns;

import java.util.regex.Pattern;

public class StringUtils {

    public static boolean validateEmail(CharSequence text) {
        Pattern pattern = Patterns.EMAIL_ADDRESS;
        return text != null && !text.toString().equals("") && pattern.matcher(text.toString().trim()).matches();
    }

    public static boolean validatePassword(CharSequence text) {
        return (text != null && !text.toString().equals("") && text.toString().length() > 6);
    }

    public static boolean validateSamePassword(CharSequence text, CharSequence text2) {
        return (text != null && !text.toString().equals("") && !text.equals(text2));
    }

}

File: app/src/main/java/com/karla/learningverbs/utils/StringUtils.kt
package com.karla.learningverbs.utils

import android.util.Patterns

object StringUtils {
    fun validateEmail(text: CharSequence?): Boolean {
        val pattern = Patterns.EMAIL_ADDRESS
        return text != null && text.toString() != "" && pattern.matcher(
            text.toString().trim { it <= ' ' }).matches()
    }

    fun validatePassword(text: CharSequence?): Boolean {
        return text != null && text.toString() != "" && text.toString().length > 6
    }

    fun validateSamePassword(text: CharSequence?, text2: CharSequence): Boolean {
        return text != null && text.toString() != "" && text != text2
    }
}